qiskit/circuit/equivalence.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gate equivalence library."""
+
+from typing import Optional, List
+from collections import namedtuple
+
+from .exceptions import CircuitError
+from .parameterexpression import ParameterExpression
+
+Key = namedtuple('Key', ['name',
```

Comment:
Why `namedtuple`s instead of objects?

Reply:
Only because `namedtuples` are hashable by default, we're only really using field aliasing at this point (their only for internal clarity). We could switch to objects in the future if needed.